app/views/submissions/browse.html.erb

```diff
@@ -14,62 +14,84 @@
   </div>
 <% end %>
 
-<% content_for :heading_buttons do %>
-  <% if @current_user.admin? %>
-    <%= link_to t('collect_submissions.collect_all'),
-                collect_all_submissions_assignment_submissions_path(
-                  @assignment.id),
-                confirm: t('collect_submissions.results_loss_warning') %>
-    <span class='menu_bar'></span>
-    <%= link_to t('browse_submissions.csv_report'),
-                download_simple_csv_report_assignment_submissions_path(
-                  @assignment) %>
-    <span class='menu_bar'></span>
-    <%= link_to t('browse_submissions.detailed_csv_report'),
-                download_detailed_csv_report_assignment_submissions_path(
-                  @assignment) %>
-    <span class='menu_bar'></span>
-    <%= link_to t('browse_submissions.subversion_repo_list'),
-                download_svn_repo_list_assignment_submissions_path(
-                  @assignment) %>
-    <span class='menu_bar'></span>
-    <%= link_to t('browse_submissions.subversion_checkout_file'),
-                download_svn_checkout_commands_assignment_submissions_path(
-                  @assignment) %>
-    <span class='menu_bar'></span>
-  <% end %>
 
-  <% # Allow TA submission collection only if all the submissions are
-     # assignment to exactly one TA. %>
-  <% if @current_user.ta? and @assignment.past_collection_date? and
-        @groupings.all? {|grouping| grouping.get_ta_names.size == 1} %>
-    <%= link_to t('collect_submissions.collect_ta'),
-                collect_ta_submissions_assignment_submissions_path(
-                  @assignment),
-                confirm: t('collect_submissions.results_loss_warning') %>
-    <span class='menu_bar'></span>
-  <% elsif @current_user.ta? and @assignment.past_collection_date? %>
-    <a id='disable' title='<%= t('collect_submissions.disable_collect') %>'>
-      <%= t('collect_submissions.collect_ta') %>
-    </a>
-    <span class='menu_bar'></span>
-  <% end %>
 
-  <% if all_assignments_marked? %>
-    <%= link_to t('browse_submissions.download_groupings_files'),
-                download_groupings_files_assignment_submissions_path(
-                  groupings: @groupings) %>
-  <% else %>
-    <%= link_to t('browse_submissions.download_groupings_files'),
-                download_groupings_files_assignment_submissions_path(
+<% @heading_buttons = [] %>
+
+<% if @current_user.admin? %>
+  <% @heading_buttons +=
+  [
+    {
+         link_text: t('collect_submissions.collect_all'),
+         link_path: collect_all_submissions_assignment_submissions_path(
+                    @assignment.id),
+      html_options: { confirm: t('collect_submissions.results_loss_warning') }
```

Comment:
Please align the keys rather than the colons (it's more consistent with other dictionaries in MarkUs).